src/test/rpc/AccountObjects_test.cpp

```diff
@@ -1032,13 +1032,152 @@ class AccountObjects_test : public beast::unit_test::suite
+            auto ptr = buf.data();
+            std::memcpy(ptr, &flags, sizeof(flags));
+            ptr += sizeof(flags);
+            std::memcpy(ptr, &fee, sizeof(fee));
+            ptr += sizeof(fee);
+            std::memcpy(ptr, issuer.data(), issuer.size());
+            ptr += issuer.size();
+            std::memcpy(ptr, &taxon, sizeof(taxon));
+            ptr += sizeof(taxon);
+            std::memcpy(ptr, &tokenSeq, sizeof(tokenSeq));
+            ptr += sizeof(tokenSeq);
+            return uint256::fromVoid(buf.data());
+        };
+
+        // this lambda function is used to check if the account_nfts method
+        // returns the correct token information. lastIndex is used to query the
+        // last marker.
+        auto compareNFTs = [&](unsigned const limit,
+                               unsigned const lastIndex,
+                               unsigned const nftsSize,
+                               const std::vector<Json::Value>& tokenIDs) {
+            Json::Value params;
+            params[jss::account] = bob.human();
+            params[jss::limit] = limit;
+            params[jss::marker] = tokenIDs[lastIndex];
+            params[jss::ledger_index] = "validated";
+            auto const resp =
+                env.rpc("json", "account_nfts", to_string(params));
+
+            if (resp[jss::result].isMember(jss::error))
+                return false;
+
+            auto const& nfts = resp[jss::result][jss::account_nfts];
+            auto nftsCount = nftsSize - lastIndex - 1 < limit
+                ? nftsSize - lastIndex - 1
+                : limit;
+
+            if (nfts.size() != nftsCount)
+                return false;
+
+            for (unsigned i = 0; i < nftsCount; i++)
+            {
+                if (nfts[i]["NFTokenID"] != tokenIDs[lastIndex + 1 + i])
+                    return false;
+            }
+
+            return true;
+        };
+
+        unsigned nftsSize = 10;
+        for (unsigned i = 0; i < nftsSize; i++)
+        {
+            env(token::mint(bob, 0));
+        }
+
+        env.close();
+
+        // save the NFTokenIDs to use later
+        std::vector<Json::Value> tokenIDs;
+        {
+            Json::Value params;
+            params[jss::account] = bob.human();
+            params[jss::ledger_index] = "validated";
+            auto const resp =
+                env.rpc("json", "account_nfts", to_string(params));
+            auto const& nfts = resp[jss::result][jss::account_nfts];
+            for (auto const& nft : nfts)
+                tokenIDs.push_back(nft["NFTokenID"]);
+        }
+
+        // test a valid marker which is equal to the third tokenID
+        BEAST_EXPECT(compareNFTs(4, 2, nftsSize, tokenIDs));
+
+        // test a valid marker which is equal to the 8th tokenID
+        BEAST_EXPECT(compareNFTs(4, 7, nftsSize, tokenIDs));
+
+        // test an unassociated marker which does not exist in the NFTokenIDs
+        {
+            Json::Value params;
```

Comment:
I'm wondering if it is better to put BEAST_EXPECT outside the lambda function. which makes the code as follows:

```C++
      auto testInvalidMarker = [&env, &bob](
                                     auto marker, char const* errorMessage) {
            Json::Value params;
            params[jss::account] = bob.human();
            params[jss::limit] = 4;
            params[jss::ledger_index] = jss::validated;
            params[jss::marker] = marker;
            Json::Value const resp =
                env.rpc("json", "account_nfts", to_string(params));
            return resp[jss::result][jss::error_message] == errorMessage;
        };

        // test an invalid marker that is not a string
        BEAST_EXPECT(testInvalidMarker(17, "Invalid field \'marker\', not string."));

        // test an invalid marker that has a non-hex character
        BEAST_EXPECT(testInvalidMarker(
            "00000000F51DFC2A09D62CBBA1DFBDD4691DAC96AD98B900000000000000000G",
            "Invalid field \'marker\'."));

        // test an unassociated marker which does not exist in the NFTokenIDs
        BEAST_EXPECT(testInvalidMarker(
            createFakeNFTMarker(bob.id(), 0x000000000, 0x00000000),
            "Invalid field \'marker\'."));

        // test an unassociated marker which exceeds the maximum value of the
        // existing NFTokenID
        BEAST_EXPECT(testInvalidMarker(
            createFakeNFTMarker(bob.id(), 0xFFFFFFFF, 0xFFFFFFFF),
            "Invalid field \'marker\'."));
```

removed `this` in testInvalidMarker's capture because it's not used.

Moving BEAST_EXPECT out is to better locate the error if some check fails. Otherwise, we can't tell where's the error directly if the BEAST_EXPECT check fails in the lambda function

Reply:
Yes, I agree.  This looks better than what I suggested.  Moving the `BEAST_EXPECT` out gives more understandable errors messages if the test fails.  Thanks!